### PR TITLE
ACCUMULO-4326 Accept Text and CharSeq in Value

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

**Comment by joshelser** [comment]
:+1:

File: core/src/main/java/org/apache/accumulo/core/data/Value.java [hunk 1]
```diff
@@ -48,6 +50,30 @@ public Value() {
...
+  /**
+   * Creates a Value using the bytes of the Text. Makes a copy, does not use the byte array from the Text.
+   *
+   * @param text
+   *          may not be null
+   *
+   * @since 1.8.0
+   */
+  public Value(Text text) {
+    this(text.getBytes(), 0, text.getLength());
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/data/Value.java]: I was not sure about copying this.  Could implement it such that if the byte array from the text is the correct length, then a copy could be avoided.   I decided against for two reasons.

 * Sometimes copying and sometimes referencing seems confusing
 * The Key constructors that take Text always copy.

Comment (joshelser) [inline core/src/main/java/org/apache/accumulo/core/data/Value.java]: IMO, you doc'ed it and there is a way for users to do a non-copy constructor with Text. That's both of the things I would be worried about :)

**Comment by keith-turner** [comment]
I pushed this.  Thanks for taking a look @joshelser

**Pull request closed without merging**